Editor/Editors/CinemachineSplineDollyEditor.cs

```diff
@@ -0,0 +1,96 @@
+#if CINEMACHINE_UNITY_SPLINES
+using System;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Splines;
+
+namespace Cinemachine.Editor
+{
+    [CustomEditor(typeof(CinemachineSplineDolly))]
+    [CanEditMultipleObjects]
+    sealed class CinemachineSplineDollyEditor : BaseEditor<CinemachineSplineDolly>
+    {
+        public override void OnInspectorGUI()
+        {
+            BeginInspector();
+            bool needWarning = false;
+            for (int i = 0; !needWarning && i < targets.Length; ++i)
+                needWarning = (targets[i] as CinemachineSplineDolly).m_Spline == null;
+            if (needWarning)
+                EditorGUILayout.HelpBox("A Path is required", MessageType.Warning);
+
+            needWarning = false;
+            for (int i = 0; !needWarning && i < targets.Length; ++i)
+                needWarning = (targets[i] as CinemachineSplineDolly).m_AutoDolly.m_Enabled 
+                    && (targets[i] as CinemachineSplineDolly).FollowTarget == null;
+            if (needWarning)
+                EditorGUILayout.HelpBox("AutoDolly requires a Follow Target", MessageType.Warning);
+
+            DrawRemainingPropertiesInInspector();
+        }
+
+        [DrawGizmo(GizmoType.Active | GizmoType.NotInSelectionHierarchy
+                | GizmoType.InSelectionHierarchy | GizmoType.Pickable, typeof(CinemachineSplineDolly))]
+        static void DrawGizmos(CinemachineSplineDolly dolly, GizmoType selectionType)
+        {
+            if (Selection.activeGameObject == dolly.VirtualCamera.gameObject)
+                DrawDollyGizmo(dolly, Color.green, 1); // GML todo: make this configurable
+        }
+
+        static void DrawDollyGizmo(CinemachineSplineDolly dolly, Color pathColor, float width)
+        {
+            var spline = dolly.m_Spline;
+            if (spline == null || spline.Spline == null || spline.Spline.Count == 0)
+                return;
+
+            var length = spline.CalculateLength();
+            var numSteps = Mathf.RoundToInt(Mathf.Clamp(length / width, 3, 1000));
```

Comment:
this 1000 also is a little ad-hoc.  Not sure of the best way to make this make sense.  The idea is to be able to draw very long paths without killing performance.